core/pta/veraison_attestation/cbor.c

```diff
@@ -0,0 +1,346 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2024, Institute of Information Security (IISEC)
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <mempool.h>
+
+#include "cbor.h"
+#include "sign.h"
+
+typedef struct {
+	UsefulBufC ubc_eat_profile;
+	int psa_client_id;
+	int psa_security_lifecycle;
+	UsefulBufC ubc_psa_implementation_id;
+	UsefulBufC ubc_measurement_type;
+	UsefulBufC ubc_signer_id;
+	UsefulBufC ubc_psa_instance_id;
+	UsefulBufC ubc_psa_nonce;
+	UsefulBufC ubc_measurement_value;
+} cbor_evidence_args;
+
+typedef struct {
+	UsefulBufC ubc_cbor_evidence;
+} cose_evidence_args;
+
+typedef struct {
+	UsefulBufC protected_header;
+	UsefulBufC aad;
+	UsefulBufC payload;
+} tbs_structure_args;
+
+void encode_cbor_evidence(
+	QCBOREncodeContext *context, UsefulBufC ubc_eat_profile,
+	const int psa_client_id, const int psa_security_lifecycle,
+	UsefulBufC ubc_psa_implementation_id, UsefulBufC ubc_measurement_type,
+	UsefulBufC ubc_signer_id, UsefulBufC ubc_psa_instance_id,
+	UsefulBufC ubc_psa_nonce, UsefulBufC ubc_measurement_value);
+void encode_cose_evidence(QCBOREncodeContext *context,
+			  UsefulBufC ubc_cbor_evidence);
+void encode_protected_header(QCBOREncodeContext *context);
+void encode_tbs_structure(QCBOREncodeContext *context,
+			  UsefulBufC protected_header, UsefulBufC aad,
+			  UsefulBufC payload);
+
+void encode_cbor_evidence_wrapper(QCBOREncodeContext *context, void *args);
+void encode_cose_evidence_wrapper(QCBOREncodeContext *context, void *args);
+void encode_protected_header_wrapper(QCBOREncodeContext *context, void *args);
+void encode_tbs_structure_wrapper(QCBOREncodeContext *context, void *args);
+
+UsefulBufC build_encoded_buffer(void (*encode_func)(QCBOREncodeContext *,
+						    void *),
+				void *encode_args);
+UsefulBufC build_cbor_evidence(UsefulBufC ubc_eat_profile, int psa_client_id,
+			       int psa_security_lifecycle,
+			       UsefulBufC ubc_psa_implementation_id,
+			       UsefulBufC ubc_measurement_type,
+			       UsefulBufC ubc_signer_id,
+			       UsefulBufC ubc_psa_instance_id,
+			       UsefulBufC ubc_psa_nonce,
+			       UsefulBufC ubc_measurement_value);
+UsefulBufC build_cose_evidence(UsefulBufC ubc_cbor_evidence);
+UsefulBufC build_protected_header(void);
+UsefulBufC build_tbs_structure(UsefulBufC protected_header, UsefulBufC aad,
+			       UsefulBufC payload);
```

Comment:
All these fonctions are local. Add `static` keyword? Remove above declarations by re-ordering the functions in the source file?

Reply:
DONE